### PR TITLE
Tell git our build directory is "safe"

### LINKED ISSUE #193: "make docker-deploy" fails with "fatal: detected dubious ownership in repository at '/veraison/build'"
### What version of the package are you using?

`HEAD -> main`

### Does this issue reproduce with the latest release?

yes

### What OS and CPU architecture are you using (`go env`)?

```shell
➜  services git:(main) ✗ go env
GO111MODULE=''
GOARCH='arm64'
GOBIN=''
GOCACHE='/Users/tho/Library/Caches/go-build'
GOENV='/Users/tho/Library/Application Support/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFLAGS=''
GOHOSTARCH='arm64'
GOHOSTOS='darwin'
GOINSECURE=''
GOMODCACHE='/Users/tho/go/pkg/mod'
GONOPROXY=''
GONOSUMDB=''
GOOS='darwin'
GOPATH='/Users/tho/go'
GOPRIVATE=''
GOPROXY='https://proxy.golang.org,direct'
GOROOT='/usr/local/go'
GOSUMDB='sum.golang.org'
GOTMPDIR=''
GOTOOLCHAIN='auto'
GOTOOLDIR='/usr/local/go/pkg/tool/darwin_arm64'
GOVCS=''
GOVERSION='go1.21.0'
GCCGO='gccgo'
AR='ar'
CC='clang'
CXX='clang++'
CGO_ENABLED='1'
GOMOD='/Users/tho/Code/github.com/veraison/services/go.mod'
GOWORK=''
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
PKG_CONFIG='pkg-config'
GOGCCFLAGS='-fPIC -arch arm64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -ffile-prefix-map=/var/folders/st/1wblnvg95bg4p9hnzc6dwjzh0000gn/T/go-build2713434598=/tmp/go-build -gno-record-gcc-switches -fno-common'
```

### What did you do?

```shell
make docker-deploy
```

### What did you expect to see?

I expected a successful build of the Docker-based deployment.

### What did you see instead?

A failed build with a bunch of repeated messages:

```shell
make[1]: Entering directory '/veraison/build/config'
fatal: detected dubious ownership in repository at '/veraison/build'
To add an exception for this directory, call:

	git config --global --add safe.directory /veraison/build
```

### PR DESCRIPTION
**Pull request by thomas-fossati**
Fix #193

### CODE REVIEW

**Review by yogeshbdeshpande — APPROVED** [review]
LGTM!

File: deployments/docker/src/builder-dispatcher [hunk 1]
```diff
@@ -29,6 +29,10 @@ function deploy() {
     mkdir -p $DEPLOY_DIR/logs
     mkdir -p $DEPLOY_DIR/utils
 
+    # tell git our build directory is "safe" -- the operating context of
+    # CVE-2022-24765 (i.e., multi-user machine) does not really apply here)
```
Comment (thomas-fossati) [inline deployments/docker/src/builder-dispatcher:33]:
```suggestion
    # CVE-2022-24765 (i.e., multi-user machine) does not really apply here
```

**Review by paulhowardarm — APPROVED** [review]
Awesome, thank you!

**Review by setrofim — APPROVED** [review]
This is probably fine as-is, as I do't imagine there is much harm in doing this on each `deploy()`, however, given that this really only needs to be configured once, this could probably be done inside the builder's dockerfile instead?

**Comment by thomas-fossati** [comment]
> [...] given that this really only needs to be configured once, this could probably be done inside the builder's dockerfile instead?

yeah, I thought about it but the relevant directory tree is created by `deploy()` and I wasn't sure whether the existence of `safe.directory` is checked by git or not.

**Pull request merged**